apps/prairielearn/src/question-servers/freeform.ts

```diff
@@ -1065,6 +1193,11 @@ export async function generate(question, course, variant_seed) {
         data,
         context,
       );
+
+      // if (resultData.params == null || resultData.correct_answers == null) {
```

Comment:
I'm not sure on these checks, my guess is that we need to either loosen all the types in `types.ts`, or tighten up the workers to return a valid type.

Reply:
I took the "tighten up types" approach in 41c9311.

Reply:
Great.